manifests/designate/ns.pp

```diff
@@ -0,0 +1,42 @@
+# Designate publishing nameserver (NS) running bind9
+class ntnuopenstack::designate::ns {
+  require ::ntnuopenstack::repo
+  include ::ntnuopenstack::designate::firewall::dns
+  include ::ntnuopenstack::designate::firewall::rndc
+
+  $api_servers = lookup('ntnuopenstack::designate::api_servers', Array[Stdlib::IP::Address]);
+  $transfer_addresses = lookup('ntnuopenstack::designate::transfer_addresses', Array[String]);
+
+  $listen_on = 'any';
+  $listen_on_v6 = 'any';
+
+  class {'dns':
+    recursion          => 'no',
+    allow_recursion    => [],
+    listen_on_v6       => false, # Overwritten by additional_options
+    localzonepath      => 'unmanaged',
+    additional_options => {
+      'listen-on'         => "port 53 { ${listen_on}; }",
+      'listen-on-v6'      => "port 53 { ${listen_on_v6}; }",
+      'auth-nxdomain'     => 'no',
+      'allow-new-zones'   => 'yes',
+      'allow-notify'      => "{ ${join($api_servers, '; ')}; }",
+      'allow-update'      => "{ ${join($api_servers, '; ')}; }",
+      'allow-transfer'    => "{ ${join($api_servers + $transfer_addresses, '; ')}; }",
```

Comment:
`transfer_addresses` bør kanskje inkludere profile::networks::infra::all for feilsøkingsformål, som @obrestad har nevnt tidligere.
Hvis vi noen gang skal bruke designate som hidden master / slave soner herifra til de vanlige NTNU-NS-ene må vi også legge de til her.

Reply:
infra:all løst i https://github.com/ntnusky/ntnuopenstack/pull/101/commits/a290f9c414d7bc5ec2c557c3c7a9cb44f339354c